src/app/modules/library/personal-library/personal-library.component.ts

```diff
@@ -80,6 +115,69 @@ export class PersonalLibraryComponent extends LibraryComponent {
   protected getDetailsComponent(): any {
     return PersonalLibraryDetailsComponent;
   }
+
+  public filterUpdated(filterValues: ProjectFilterValues = null): void {
+    super.filterUpdated(filterValues);
+    this.filteredProjects = this.filteredProjects.filter(
+      (project) => project.tags.includes('archived') == this.showArchived
+    );
+  }
+
+  protected switchActiveArchivedView(): void {
+    this.filterUpdated();
+    this.unselectAllProjects();
+  }
+
+  pageChange(event?: PageEvent, scroll?: boolean): void {
+    super.pageChange(event, scroll);
+    this.unselectAllProjects();
+  }
+
+  protected projectSelectionChanged(event: any): void {
+    if (event.checked) {
+      this.selectedProjects.update((selectedProjects) => {
+        selectedProjects.push(event.project);
+        return selectedProjects;
+      });
+    } else {
+      this.selectedProjects.update((selectedProjects) => {
+        selectedProjects.splice(selectedProjects.indexOf(event.project), 1);
+        return selectedProjects;
+      });
+    }
+  }
+
+  protected archiveSelectedProjects(archive: boolean): Subscription {
+    return this.archiveProjectService[archive ? 'archiveProjects' : 'unarchiveProjects'](
+      this.selectedProjects()
+    ).subscribe({
+      next: () => {
+        this.archiveProjectService.showArchiveProjectsSuccessMessage(
+          this.selectedProjects(),
+          archive
+        );
+        this.archiveProjectService.updateProjectsArchivedStatus(this.selectedProjects(), archive);
+        this.unselectAllProjects();
+      },
+      error: () => {
+        this.archiveProjectService.showArchiveProjectErrorMessage(archive);
+      }
+    });
+  }
+
+  private unselectAllProjects(): void {
+    this.selectAllProjects(false);
+  }
+
+  protected selectAllProjects(selectAll: boolean): void {
+    this.selectedProjects.update(() => (selectAll ? this.getProjectsInView() : []));
```

Comment:
Use set() here? update() is used when you want to do something with the current value of selectedProjects.
```suggestion
    this.selectedProjects.set(() => (selectAll ? this.getProjectsInView() : []));
```